lib/stats-collector.js

```diff
@@ -1,19 +1,21 @@
 'use strict';
 
 /**
- * Initializes test statistics collector object.
+ * Creates test statistics collector object.
  *
  * Provides stats such as test duration,
  * number of tests passed / failed etc.
  *
  * @public
  * @memberof Mocha
  * @param {Runner} runner
- * @return {StatsCollector}
- *
- * Reporter statistics.
+ * @return {stats}
+ */
+
+/**
+ * {stats} object properties
  *
- * @typedef {Object} StatsCollector
+ * @typedef {Object} stats
```

Comment:
Revert this change. We're documenting what we're creating. As the function is named `createStatsCollector()`, the `@typedef` should match - hence:
`@typedef {Object} StatsCollector`.

Reply:
`@typedef` docblock should **preceed** function docblock.

Invert them so function docblock is paired with the function.

Reply:
@plroebuck I have now reverted this change.